### PR TITLE
etcdserver: serializable transactions

### PR DESCRIPTION
**Pull request by heyitsanthony**
Support case where txn doesn't have to go through quorum.

### CODE REVIEW

**Comment by gyuho** [comment]
lgtm.

File: etcdserver/v3demo_server.go [hunk 1]
```diff
@@ -94,6 +98,20 @@ func (s *EtcdServer) Txn(ctx context.Context, r *pb.TxnRequest) (*pb.TxnResponse
 	return result.resp.(*pb.TxnResponse), result.err
 }
 
+func isTxnSerializable(r *pb.TxnRequest) bool {
+	for _, u := range r.Success {
```
Comment (xiang90) [inline etcdserver/v3demo_server.go]: we need to ensure there is no write/delete in txn too, right?

Comment (xiang90) [inline etcdserver/v3demo_server.go]: oh... nvm

**Comment by xiang90** [comment]
lgtm

File: etcdserver/v3demo_server.go (hunk 1 above)
Comment (heyitsanthony) [inline etcdserver/v3demo_server.go]: Yep. That's what the `r == nil` does.

**Comment by heyitsanthony** [comment]
merging on greenlights

**Comment by heyitsanthony** [comment]
failing on unrelated race, merging anyway

**Pull request merged**